src/en/TravisTranslations.lua

```diff
@@ -0,0 +1,226 @@
+		local KEY_ID = LANGUAGE_FILTER + i
+		return CheckboxFilter(KEY_ID, v)
+	end)),
+	FilterGroup("Genre", map(GENRE_VALUES, function(v,i) 
+		local KEY_ID = GENRE_FILTER + i
+		return CheckboxFilter(KEY_ID, v)
+	end))
+}
+
+local encode = Require("url").encode
+
+local text = function(v)
+	return v:text()
+end
+
+local function shrinkURL(url)
+	return url:gsub("^.-travistranslations%.com", "")
+end
+
+local function expandURL(url)
+	return baseURL .. url
+end
+
+local function getPassage(chapterURL)
+	local content = GETDocument(expandURL(chapterURL)):selectFirst("main#primary")
+	-- Removes the title and makes it an H1 for consistant custom CSS.
+	local title = content:selectFirst("h2"):text()
+	local chap = content:selectFirst(".reader-content")
+	chap:child(0):before("<h1>" .. title .. "</h1>")
+	-- Remove empty <p> tags
+	local toRemove = {}
+	chap:traverse(NodeVisitor(function(v)
+		if v:tagName() == "p" and v:text() == "" then
+			toRemove[#toRemove+1] = v
+		end
+		if v:hasAttr("border") then
+			v:removeAttr("border")
+		end
+	end, nil, true))
+	for _,v in pairs(toRemove) do
+		v:remove()
+	end
+	return pageOfElem(chap, true)
+end
+
+local function parseNovel(novelURL, loadChapters)
+	local doc = GETDocument(baseURL .. novelURL)
+	local content = doc:selectFirst("main#primary")
+
+	local info = NovelInfo {
+		title = content:selectFirst("h1#heading"):attr("title"),
+		imageURL = "https:" .. content:selectFirst("img"):attr("data-src"),
+		status = ({
+			Completed = NovelStatus.COMPLETED,
+			Ongoing = NovelStatus.PUBLISHING
+		})[content:selectFirst(".grid.grid-cols-2.gap-3.my-2"):child(3):selectFirst("strong"):text()],
+		description = table.concat(map(doc:selectFirst('div[property="description"]'):select("p"), text), '\n'),
+		artists = { "Translator: Travis Translations" },
+		genres = map(content:selectFirst("ul.space-x-1.space-y-2"):select("a"), text),
+		tags = map(content:selectFirst("ul.flex.flex-wrap.my-2"):select("a"), text)
+	}
+
+	if loadChapters then
+		local chapters = (map(content:selectFirst('div[x-show="tab === \'toc\'"]'):select("li"), function(v, i)
+			-- This is to ignore the premium chapter, those have a lock icon in their anchor.
+			local firstPremChapter = v:selectFirst("i.i-lock")
+			if firstPremChapter ~= nil then return end
+			
+			local a = v:selectFirst("a")
+			return NovelChapter {
+				order = i,
+				title = a:child(0):text(),
+				link = shrinkURL(a:attr("href")),
+				release = a:child(1):text()
+			}
+		end))
+
+		info:setChapters(AsList(chapters))
+	end
+	return info
+end
+
+local function parseListing(listingURL)
+	local doc = GETDocument(listingURL)
+	return map(doc:select("li.group"), function(v)
+		local a = v:selectFirst("a")
+		return Novel {
+			title = a:attr("title"),
+			link = shrinkURL(a:attr("href")),
+			imageURL = "https:" .. v:selectFirst("img"):attr("data-src")
+		}
+	end)
+end
+
+local function getSearch(data)
+	local query = data[QUERY]
+	local page = data[PAGE] + 1
```

Comment:
Is the +1 there to avoid page 0? If so put a todo for when the startIndex is finally put to 1. Or perhaps check if it is 0 and then set it to one.

Reply:
This actually fixes most of the duplication problems in the search too, and honestly i can put a todo to change it back one day.